pkg/controller/directvolumemigration/rsync.go

```diff
@@ -1211,21 +1217,44 @@ func (t *Task) createPVProgressCR() error {
 					Namespace: migapi.OpenshiftMigrationNamespace,
 				},
 				Spec: migapi.DirectVolumeMigrationProgressSpec{
-					ClusterRef: t.Owner.Spec.SrcMigClusterRef,
-					PodRef: &corev1.ObjectReference{
-						Namespace: ns,
-						Name:      fmt.Sprintf("directvolumemigration-rsync-transfer-%s", vol.Name),
-					},
+					ClusterRef:   t.Owner.Spec.SrcMigClusterRef,
+					PodNamespace: ns,
+					PodSelector:  GetRsyncPodSelector(vol.Name),
 				},
-				Status: migapi.DirectVolumeMigrationProgressStatus{},
+			}
+			existingDvmp := migapi.DirectVolumeMigrationProgress{}
+			shouldDelete := false
+			// Make sure existing DVMP CRs which don't have required fields are deleted
+			err := t.Client.Get(context.TODO(), types.NamespacedName{Name: dvmp.Name, Namespace: dvmp.Namespace}, &existingDvmp)
+			if err != nil {
+				if !k8serror.IsNotFound(err) {
+					return err
+				}
+			}
+			if existingDvmp.Name != "" && existingDvmp.Namespace != "" {
```

Comment:
put this in a separate function like `deleteDVMPForSmoothUpgrade` to make it explicit and not pollute this function?

Reply:
+1, will do